src/legacy/ui/public/search_bar/directive/index.js

```diff
@@ -27,10 +27,33 @@ const app = uiModules.get('app/kibana', ['react']);
 app.directive('searchBar', (reactDirective, localStorage) => {
   return reactDirective(
     wrapInI18nContext(SearchBar),
-    undefined,
+    [
+      ['query', { watchDepth: 'reference' }],
+      ['store', { watchDepth: 'reference' }],
+      ['intl', { watchDepth: 'reference' }],
+
+      ['onQuerySubmit', { watchDepth: 'reference' }],
+      ['onFiltersUpdated', { watchDepth: 'reference' }],
+      ['onRefreshChange', { watchDepth: 'reference' }],
+
+      ['indexPatterns', { watchDepth: 'collection' }],
+      ['filters', { watchDepth: 'collection' }],
+
+      'appName',
+      'screenTitle',
+      'showFilterBar',
+      'showQueryBar',
+      'showDatePicker',
+      'dateRangeFrom',
+      'dateRangeTo',
+      'isRefreshPaused',
+      'refreshInterval',
+      'disableAutoFocus',
+      'showAutoRefreshOnly',
```

Comment:
I feel like maybe we shouldn't list the attributes that simply use the default config. I can see this list becoming out of date quickly if someone makes a change to the props and forgets to update this file.

Reply:
@Bargs  unfortunately we should =(.

Reply:
@alexwizp can you explain why, for the sake of my own education?

Reply:
@Bargs sure, but firstly let's debug a little the following code from ngReact:

![image](https://user-images.githubusercontent.com/20072247/56416130-983a0100-6298-11e9-8501-0e1c2807f280.png)

I highlighted  the value for the `reactComponent.propTypes ` to show you why we have this issue. It's `undefined ` because not all developers declare **propTypes** for TS Components like e.g. happened in `SearchBar`.  

From the other hand we pass **undefined** as a second argument to `reactDirective` function. 

And it's a tricky moment. As you see we don't have data to add watchers.  

So we have 2 ways how to fix it: 
1) declare `propTypes` 
2) do like in my PR.  - I decided to use this way because only using it I can setup right value for watch-depth property for each fields

But from my point of view we should always implement  1 + 2.

Reply:
Oh... looks like I'm not fully right. In lines 264-270 we read props from arguments if `!props.length`.
But again to set the correct value for watch-depth property we have only one legal way which I implemented in my PR.

Reply:
Thanks, I understand now. The props listed here are all or nothing, ngReact will not look at the angular attributes if any props are listed in this config. Since it is very important that this list stay in sync with the list of props in search_bar.tsx, could you add a comment [here](https://github.com/elastic/kibana/blob/ed795d28ee780928b307f2271ee6acd9404d7485/src/legacy/ui/public/search_bar/components/search_bar.tsx#L41) reminding any future devs that changes to the props need to be made in the directive config as well?